pkg/proxy/util/utils.go

```diff
@@ -255,26 +255,26 @@ func LogAndEmitIncorrectIPVersionEvent(recorder record.EventRecorder, fieldName,
 	}
 }
 
-// FilterIncorrectIPVersion filters out the incorrect IP version case from a slice of IP strings.
-func FilterIncorrectIPVersion(ipStrings []string, ipfamily v1.IPFamily) ([]string, []string) {
-	return filterWithCondition(ipStrings, (ipfamily == v1.IPv6Protocol), utilnet.IsIPv6String)
+// MapIPsToIPFamily maps a slice of IPs to their respective IP families (v4 or v6)
+func MapIPsToIPFamily(ipStrings []string, isCIDR bool) map[v1.IPFamily][]string {
+	ipFamilyMap := map[v1.IPFamily][]string{}
+	for _, ip := range ipStrings {
+		ipFamily := getIPFamilyFromIP(ip, isCIDR)
```

Comment:
Don't we have some error handling in there SOMEWHERE?  What if I pass "gobbledygook" - neither v4 nor v6?

Reply:
Looks like we were handling all error cases as IPV4 with this check -> ipfamily == v1.IPv6Protocol.

I'll check and add some error handling in this path.

Reply:
Error handling when IPs are incorrect is not trivial. I have added net.ParseIP and net.ParseCIDR in my Map functions to handle only correct IPs and CIDRs. If we want more robust error handling, I can take it up in the next clean up activity.